Parser/pegen.c

```diff
@@ -2383,28 +2383,81 @@ _PyPegen_concatenate_strings2(Parser *p, asdl_expr_seq *strings,
         return _PyAST_Constant(res, NULL, lineno, col_offset, end_lineno, end_col_offset, p->arena);
     }
 
-    asdl_expr_seq* values = _Py_asdl_expr_seq_new(n_elements, p->arena);
-    if (values == NULL) {
+    if (unicode_string_found && !f_string_found && len == 1) {
+        return asdl_seq_GET(strings, 0);
+    }
```

Comment:
Is there some special handling of f-strings in here that prevents us from just cheking for `len == 1`?

Reply:
`f'{1}'` produces, or should produce, `JoinedStr(values=[FormattedValue(value=Constant(value=1), conversion=-1)])`, so yes.

Currently, we seem to be producing `JoinedStr(values=[FormattedValue(value=Constant(value=1), conversion=-1), Constant(value='')])`. I am not sure if that should be fixed in grammar, or in this function. If we fix it here, by skipping empty strings, we can just do `len == 1`.

Reply:
I'm fixing this exact problem in #164.